backend/src/load.rs

```diff
@@ -0,0 +1,407 @@
+                continue;
+            }
+            file_count += 1;
+
+            let filename = entry.file_name().into_string().unwrap();
+            let mut file = File::open(entry.path())?;
+            let mut file_contents = String::new();
+            // Skip files whose size is 0.
+            if file.read_to_string(&mut file_contents)? == 0 {
+                no_times += 1;
+                continue;
+            }
+
+            let contents = match json::parse(&file_contents) {
+                Ok(json) => json,
+                Err(err) => {
+                    println!("Failed to parse JSON for {}: {:?}", filename, err);
+                    no_times += 1;
+                    continue;
+                }
+            };
+            if contents["times"].is_empty() {
+                no_times += 1;
+                continue;
+            }
+
+            /// Parse date from JSON header in file contents.
+            fn parse_from_header(date: &str) -> Result<NaiveDateTime> {
+                // TODO: Determine correct format of header date and move into
+                // variable/constant
+                NaiveDateTime::parse_from_str(date, "%c %z")
+                    .map_err(|err| err.into())
+            }
+
+            /// Parse date from filename.
+            fn parse_from_filename(filename: &str) -> Result<NaiveDateTime> {
+                let date_str = &filename[(filename.find("--").unwrap() + 2)..filename.find(".json").unwrap()];
+
+                // Ignore seconds: they aren't key to the data processing, and
+                // parsing them requires logic to replace them with 0 when
+                // they are absent.
+                NaiveDateTime::parse_from_str(date_str, "%Y-%m-%d-%H-%M")
+                    .map_err(|err| err.into())
+            }
+
+            let header = &contents["header"];
+            let date = header["date"].as_str().unwrap();
+            let date = parse_from_header(&date).or_else(|_| parse_from_filename(&filename))?;
+
+            let test_name = &filename[0..filename.find("--").unwrap()];
+
+            let times = &contents["times"];
+            if test_name == "rustc" {
+                data_rustc.push(Data::new(date, &header, times, true));
+
+                for timing in times.members() {
+                    let crate_name = timing["crate"].as_str().unwrap().to_string();
+                    crate_list.insert(crate_name);
+                }
+            } else {
+                benchmarks.insert(test_name.to_string());
+                let index = data_benchmarks.iter().position(|benchmark: &Data| benchmark.date == date);
+                if let Some(index) = index {
+                    c_benchmarks_add += 1;
+                    let crate_name = times[0]["crate"].as_str().unwrap();
+                    data_benchmarks[index].by_crate.insert(test_name.to_string(),
+                        make_times(times, false).remove(crate_name).unwrap());
```

Comment:
I'm not sure what the original intended purpose of this code was. However, I noticed that this means we overwrite existing data in these cases. None of these are recent, so it may not be too big a deal, but wanted to make sure we were aware of this, since it might lead to data loss in the future as well.
```
hyper.0.5.0 in hyper.0.5.0--2015-05-25-14-31-35.json
helloworld in helloworld--2015-05-25-14-31-35.json
html5ever-2015-05-15 in html5ever-2015-05-15--2015-05-25-14-31-35.json
regex.0.1.30 in regex.0.1.30--2015-05-25-14-31-35.json
rust-encoding.0.2.32 in rust-encoding.0.2.32--2015-05-25-15-45-57.json
regex-macros.0.1.30 in regex-macros.0.1.30--2015-05-25-14-31-35.json
```